app/src/main/java/org/hackillinois/android/view/navigationdrawer/ScannerFragment.kt

```diff
@@ -0,0 +1,195 @@
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        viewModel = ViewModelProviders.of(this).get(ScannerViewModel::class.java)
+        viewModel.init()
+        viewModel.eventsListLiveData.observe(this, Observer { updateEventList(it) })
+
+        // We will only observe the change in lastScanWasSuccessful, as if that changes,
+        // so will the scanned userId
+        viewModel.lastScanWasSuccessful.observe(this, Observer { processLastScan(it) })
+
+        viewModel.shouldDisplayOverrideSwitch.observe(this, Observer { updateOverrideSwitchVisibility(it) })
+    }
+
+    /**
+     * Make the switch invisible when not relevant to the event.
+     */
+    private fun updateOverrideSwitchVisibility(it: Boolean?) {
+        staffOverrideSwitch.visibility = when (it) {
+            true -> AdapterView.VISIBLE
+            false -> AdapterView.GONE
+            null -> AdapterView.GONE
+        }
+    }
+
+    /**
+     * Called when scanner scans something.
+     */
+    private var onQrCodeScan: BarcodeCallback = object : BarcodeCallback {
+        override fun possibleResultPoints(resultPoints: MutableList<ResultPoint>?) {
+        }
+
+        override fun barcodeResult(result: BarcodeResult) {
+            // Prevent duplicate scans
+            if (result.text == null || result.text == lastSuccessfullyScannedUser) {
+                return
+            }
+
+            if (lastSuccessfullyScannedUser == "") {
+                lastSuccessfullyScannedUser = result.text
+            }
+
+            Log.d("ScanEvent", "Scanned text: ${result.text}")
+            Toast.makeText(context, lastSuccessfullyScannedUser, Toast.LENGTH_LONG).show()
+
+            // Event that user is being scanned in for
+            var eventName: String = eventListView.selectedItem.toString()
+
+            // User's ID
+            var userId: String = getUserIdFromQrString(lastSuccessfullyScannedUser)
+
+
+            Log.d("ScanEvent", "User ID: ${userId}")
+            Log.d("ScanEvent", "Event: ${eventName}")
+
+            // Check-in is a special event in the HackIllinois API
+            if (eventName == "Check In") {
+                var override = staffOverrideSwitch.isChecked
+                var hasCheckedIn = true
+                var hasPickedUpSwag = true
+                var checkIn = CheckIn(userId, override, hasCheckedIn, hasPickedUpSwag)
+
+                viewModel.checkInUser(checkIn)
+
+            } else {
+                var userEventPair = UserEventPair(eventName, userId)
+
+                viewModel.markUserAsAttendingEvent(userEventPair)
+            }
+        }
+    }
+
+    /**
+     * A callback that is called when the status of the last scan request changes.
+     */
+    fun processLastScan(lastScanWasSuccessful: Boolean?) {
+        when (lastScanWasSuccessful) {
+            false -> {
+                Toast.makeText(context, "Try again!", Toast.LENGTH_LONG).show()
```

Comment:
We're using Snackbars instead of Toasts in other places in the app

Reply:
Switching over, but `ToolsFragment` and `LoginActivity` have a ton of Toasts from a quick grep of the code base.